tests/subset.rs

```diff
@@ -64,30 +77,42 @@ fn new_network<A, F>(
     good_num: usize,
     bad_num: usize,
     adversary: F,
-) -> TestNetwork<A, Subset<NodeId, u8>>
+) -> VirtualNet<Subset<NodeId, u8>, A>
 where
     A: Adversary<Subset<NodeId, u8>>,
-    F: Fn(BTreeMap<NodeId, Arc<NetworkInfo<NodeId>>>) -> A,
+    F: Fn() -> A,
 {
     // This returns an error in all but the first test.
     let _ = env_logger::try_init();
 
-    let new_subset =
-        |netinfo: Arc<NetworkInfo<NodeId>>| Subset::new(netinfo, 0).expect("new Subset instance");
-    TestNetwork::new(good_num, bad_num, adversary, new_subset)
+    let mut rng = rand::thread_rng();
+
+    let size = good_num + bad_num;
+
+    let (net, _) = NetBuilder::new(0..size as u16)
+        .num_faulty(bad_num as usize)
+        .message_limit(10_000 * size as usize)
+        .no_time_limit()
+        .adversary(adversary())
+        .using(move |node_info: NewNodeInfo<_>| {
+            Subset::new(Arc::new(node_info.netinfo), 0).expect("new Subset instance")
+        })
+        .build(&mut rng)
+        .expect("Could not construct test network.");
+    net
 }
 
 #[test]
 fn test_subset_3_out_of_4_nodes_propose() {
     let proposed_value = Vec::from("Fake news");
-    let proposing_ids: BTreeSet<NodeId> = (0..3).map(NodeId).collect();
+    let proposing_ids: BTreeSet<NodeId> = (0..3).map(|node_id| node_id).collect();
```

Comment:
Is that `map` call redundant? (Also below.)

Reply:
Indeed - removed!